main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2580,9 +2580,9 @@ object Defaults extends BuildCommon {
   private[sbt] def jnone[A]: Optional[A] = none[A].toOptional
   def compileAnalysisSettings: Seq[Setting[_]] = Seq(
     previousCompile := {
-      val setup = compileIncSetup.value
+      val analysisFile = compileAnalysisFile.value
```

Comment:
Nice. Could you add a comment saying avoid `compileIncSetup` since it would trigger upstream compilation, plz?

Reply:
Done!